### ISSUE TITLE
onExit isn't happening

### ISSUE THREAD

**Issue by david-buck** [gianlucacandiotti/react-router-transitions#1]
It seems like the exiting route is destroyed before it can be animated out, so handleExit never fired. I have messed around but been unable to get it working.

**Comment by gianlucacandiotti** [comment]
Yeah a little more work had to be done in order for it to function properly. I'll push some fixes later on today.

**Comment by gianlucacandiotti** [comment]
@david-buck I already pushed some code that fixes it. Tell me if everything works correctly. I'm not an expert using transitions or react-router so maybe this isn't the best possible way to do it. If you find a better way in the future I'd appreciate some feedback or a pr!
